src/test/operators/jit_operator/jit_aware_lqp_translator_test.cpp

```diff
@@ -0,0 +1,387 @@
+    ASSERT_NE(jit_operator_wrapper, nullptr);
+  }
+
+  {
+    predicate_node_1->set_scan_type(ScanType::IndexScan);
+    const auto jit_operator_wrapper =
+        std::dynamic_pointer_cast<JitOperatorWrapper>(lqp_translator.translate_node(predicate_node_2));
+    ASSERT_EQ(jit_operator_wrapper, nullptr);
+  }
+}
+
+TEST_F(JitAwareLQPTranslatorTest, InputColumnsAreAddedToJitReadTupleAdapter) {
+  // The query reads two columns from the input table. These input columns must be added to the JitReadTuples adapter to
+  // make their data accessible by other JitOperators.
+  const auto jit_operator_wrapper = translate_query("SELECT a, b FROM table_b WHERE a > 1");
+  const auto jit_operators = jit_operator_wrapper->jit_operators();
+  ASSERT_EQ(jit_operators.size(), 4u);
+
+  // Check that the first operator is in fact a JitReadTuples instance
+  const auto jit_read_tuples = std::dynamic_pointer_cast<JitReadTuples>(jit_operators[0]);
+  ASSERT_NE(jit_read_tuples, nullptr);
+
+  // There should be two input columns
+  const auto input_columns = jit_read_tuples->input_columns();
+  ASSERT_EQ(input_columns.size(), 2u);
+
+  ASSERT_EQ(input_columns[0].column_id, ColumnID{0});
+  ASSERT_EQ(input_columns[0].tuple_value.data_type(), DataType::Int);
+  ASSERT_EQ(input_columns[0].tuple_value.is_nullable(), true);
+
+  ASSERT_EQ(input_columns[1].column_id, ColumnID{1});
+  ASSERT_EQ(input_columns[1].tuple_value.data_type(), DataType::Float);
+  ASSERT_EQ(input_columns[1].tuple_value.is_nullable(), true);
+}
+
+TEST_F(JitAwareLQPTranslatorTest, LiteralValuesAreAddedToJitReadTupleAdapter) {
+  // The query contains two literals. Literals are treated like values read from a column inside the operator pipeline.
+  // The JitReadTuples adapter is responsible for making these literals available from within the pipeline.
+  const auto jit_operator_wrapper = translate_query("SELECT a, b FROM table_b WHERE a > 1 AND b > 1.2");
+  const auto jit_operators = jit_operator_wrapper->jit_operators();
+  ASSERT_EQ(jit_operators.size(), 4u);
+
+  // Check that the first operator is in fact a JitReadTuples instance
+  const auto jit_read_tuples = std::dynamic_pointer_cast<JitReadTuples>(jit_operators[0]);
+  ASSERT_NE(jit_read_tuples, nullptr);
+
+  // There should be two literals read from the query
+  const auto input_literals = jit_read_tuples->input_literals();
+  ASSERT_EQ(input_literals.size(), 2u);
+
+  ASSERT_TRUE(input_literals[0].value == AllTypeVariant(1));
+  ASSERT_EQ(input_literals[0].tuple_value.data_type(), DataType::Int);
+  ASSERT_EQ(input_literals[0].tuple_value.is_nullable(), false);
+
+  ASSERT_TRUE(input_literals[1].value == AllTypeVariant(1.2f));
+  ASSERT_EQ(input_literals[1].tuple_value.data_type(), DataType::Float);
+  ASSERT_EQ(input_literals[1].tuple_value.is_nullable(), false);
+}
+
+TEST_F(JitAwareLQPTranslatorTest, SelectedColumnsAreOutputInCorrectOrder) {
```

Comment:
Any reason that this isn't three independent tests?

Reply:
They are now\